### PR TITLE
Implementation of new modifiers for  issue #11367

### PR DESCRIPTION
**Pull request by Mtjpp**
Describe the changes you have made here: 

Implementation for issue: https://github.com/JabRef/jabref/issues/11367

 I have implemented new modifiers that can be used when renaming a PDF file used in file name format based on this comment: https://github.com/JabRef/jabref/issues/11367#issuecomment-2412533435 .

List of implemented modifiers:
Camel
CamelN
VeryShortTitle
ShortTitle

User documentation needs to be updated:
https://docs.jabref.org/setup/citationkeypatterns#modifiers

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by koppor — CHANGES_REQUESTED** [review]
First impression: Looks good.

There should be tests for the use of `veryhorttitle` etc. I think, you will find the existing tests and then you can refine them.

**Comment by Mtjpp** [comment]
@koppor 
Hello I have implemented unit tests can u review it please ?

File: src/main/java/org/jabref/logic/formatter/casechanger/CamelFormatter.java
Comment (koppor) [inline src/main/java/org/jabref/logic/formatter/casechanger/CamelFormatter.java:17]: Did you look at the other formatters for inspriation?

```suggestion
        return "Camel case";
```

File: src/main/java/org/jabref/logic/formatter/casechanger/CamelNFormatter.java
Comment (koppor) [inline src/main/java/org/jabref/logic/formatter/casechanger/CamelNFormatter.java:17]:
```suggestion
        return Localization.lang("Camel case - n letters max");
```

**Review by koppor — CHANGES_REQUESTED** [review]
You can remove the tests, because the functionality of the called formatters are already tested (otherwise, I need to tell you: use `@ParameterizedTest`, do not prefix tests with `test_` and we would need another tough round here)

I am sorry that I wrote "I think, you will find the existing tests and then you can refine them.". Here is the pointer: Add before  `org.jabref.logic.citationkeypattern.BracketedPatternTest#expandBracketsWithAuthorStartingWithBrackets` 

1. Add a new `@ParameterizedTest` with `@CsvSource`. You see at other JabRef code how it works (use Ctrl+Shift+F).
2. Add tests for a variant of the cases mentioned at https://github.com/JabRef/jabref/issues/11367#issuecomment-2412533435. Use field `TITLE` instead of `USER_CUSTOM_TITLE_FIELD`.

File: src/main/java/org/jabref/logic/formatter/casechanger/CamelNFormatter.java
Comment (Mtjpp) [reply src/main/java/org/jabref/logic/formatter/casechanger/CamelNFormatter.java:17]: sure i can work on the tests u mentioned but I would keep the ones I have implemented

**Comment by koppor** [comment]
Fixes #11367

**Comment by koppor** [comment]
I have no push rights. Will let it though and add the CHANGELOG.md entry after the merge.

**Review by koppor — APPROVED** [review]

**Comment by ThiloteE** [comment]
@Mtjpp would you mind updating the documentation? I was looking at it, but you know the code best...

**Pull request merged**
